libsql-wal/src/segment/mod.rs

```diff
@@ -0,0 +1,150 @@
+#![allow(dead_code)]
+use std::mem::size_of;
+use std::num::NonZeroU64;
+
+use memoffset::offset_of;
+use zerocopy::byteorder::little_endian::{U32, U64};
+use zerocopy::AsBytes;
+
+use crate::error::{Error, Result};
+
+pub mod list;
+pub mod sealed;
+
+#[repr(C)]
+#[derive(Debug, zerocopy::AsBytes, zerocopy::FromBytes, zerocopy::FromZeroes, Clone, Copy)]
+pub struct SegmentHeader {
+    start_frame_no: U64,
+    last_commited_frame_no: U64,
+    db_size: U32,
```

Comment:
Why is db_size only 32-bit number? Isn't 2GB too small?

Reply:
Db size in SQLite terminology is in pages